frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -2344,6 +2346,33 @@ func validateBaseImagePlatform(name string, expected, actual ocispecs.Platform,
 	}
 }
 
+func validateNoSecretKey(key string, location []parser.Range, lint *linter.Linter) {
+	// Check for either full value or first/last word.
+	// Examples: api_key, DATABASE_PASSWORD, GITHUB_TOKEN, secret_MESSAGE, AUTH
+	// Case insensitive.
+	secretTokens := []string{
+		"apikey",
+		"auth",
+		"credential",
+		"credentials",
+		"key",
+		"password",
+		"pword",
+		"passwd",
+		"secret",
+		"token",
+	}
+
```

Comment:
```
pattern := `(?i)(?:_|^)(?:`+strings.Join([], "|")+`)(?:_|$)`
```

Should be equivalent and more optimized matching than this scan of all tokens. Initialize with `sync.Once` if changed.